Tests/GRPCCoreTests/MetadataTests.swift

```diff
@@ -314,4 +314,34 @@ struct MetadataTests {
       )
     }
   }
+
+  @Suite("Description")
+  struct Description {
+    let metadata = Metadata([
+      "key1": "value1",
+      "key2": "value2",
+      "key-bin": .binary([1, 2, 3]),
+    ])
```

Comment:
This is using the `init` from a `Sequence`, so we're creating a dictionary then iterating the sequence of key-value pairs in the dictionary. Dictionary iteration isn't by insertion order meaning that the metadata may be ordered differently (which is why a test failed).

If you do: `let metadata: Metadata = [ .... ]` then you'll go via the `init` for `ExpressibleByDictionaryLiteral` which will use the correct order.